### DISCUSSION TITLE
JIT mode still broken: does not generate classes

### DISCUSSION THREAD

**Discussion by TomDeSmet** [tailwindlabs/tailwindcss#6187] [Help]
Hi. Is there a definitive fix for Tailwind JIT not generating classes? A lot of GIT issues about it are closed, but I'm still experiencing issues with it. With Webpack as well as with Vite. 

I already tried the docs recommendations about making the purge list paths more detailed and the `TAILWIND_MODE=watch`  option, but the problems still exists.

Is this still a known bug in the JIT engine that's being actively worked on?

**Comment by adamwathan** [comment]
_Converting to a discussion since this does not use the issue template or include any of the required information._

Can you provide a reproduction? I can't get it to not generate classes. For example tailwindcss.com is built with JIT mode and the classes have definitely been generated there.

**Comment by TomDeSmet** [comment]
I made a condensed git repo of my project here: https://github.com/TomDeSmet/tailwind-jit-issue
It's more or less a basic Vite + Tailwind setup.
I configured the purge list targeting files directly instead of with a wildcard.
I didn't include `TAILWIND_MODE=watch` as this didn't work for me either.

If you run it with `npm run dev` and add `p-8` or `text-green-500` to the `p` tag in `app.vue`. It doesn't do anything unless you save the Tailwind config file again.

**Comment by adamwathan** [comment]
This is helpful and I can reproduce, thanks! Will dig into it.

**Comment by adamwathan** [comment]
So this looks like a bug in Vite that we actually reported back in June, it doesn't respect PostCSS dependency messages for .vue files:

https://github.com/vitejs/vite/issues/3717

Thanks for the reminder, will try to push things forward there.

**Comment by adamwathan** [comment]
Testing again, one thing that's weird is in your purge config you have `app.vue` but the file is `App.vue` on the file system. Using Vue 3 this actually fixes the problem completely, but in Vue 2 it still doesn't seem to work.

Here's a Vue 3 project where all of this stuff works perfectly:

https://github.com/adamwathan/vite-issue-repro

Talked to Evan about and sent him the reproduction, he says this is an issue with `vite-plugin-vue2`, and unfortunately that's a totally third party thing and Vite doesn't support Vue 2 natively. I've opened an issue there for you, hopefully someone is willing to help out:

https://github.com/underfin/vite-plugin-vue2/issues/153

**Comment by TomDeSmet** [comment]
Thanks for figuring this out! Time to move on to Vue 3 I think.

**Comment by filipesmedeiros** [comment]
i just want to come in here and say that taking the time to debug other people's problems like this is really freaking cool of you @adamwathan 

love tailwind!

**Comment by hivokas** [comment]
@adamwathan is there a way to make `TAILWIND_MODE=watch` work in such scenario (https://github.com/underfin/vite-plugin-vue2/issues/153)?

**Comment by Acccent** [comment]
After much struggling with the same issue despite using Vue 3, I realized the problem was coming from postcss-each, I thought I'd mention this here.

My `postcss.config.js` file looks like this:
```js
module.exports = {
  plugins: {
    'postcss-import': {},
    'postcss-each': {
      plugins: {
        beforeEach: [
          require('tailwindcss/nesting'),
          require('tailwindcss'),
          require('autoprefixer'),
        ],
      },
    },
  },
};
```
 (I put Tailwind in `beforeEach` after reading this issue: #7362 – doesn't a lot of sense to me but it was the only way to get it to work)

Removing the `postcss-each` plugin and putting Tailwind and the rest at the top level enables proper HMR.

(I guess I'm gonna have to do a lot of copy-pasting – dreading the font-face rules – does anyone know of a good alternative to write some loops in css?)
